src/frontend/src/tests/utils/derived.utils.ts

```diff
@@ -0,0 +1,125 @@
+import { tokenWithFallbackAsIcToken } from '$icp/derived/ic-token.derived';
+import {
+	btcAddressMainnet,
+	btcAddressTestnet,
+	ethAddress,
+	ethAddressCertified,
+	ethAddressNotCertified,
+	ethAddressNotLoaded
+} from '$lib/derived/address.derived';
+import { authIdentity, authNotSignedIn, authSignedIn } from '$lib/derived/auth.derived';
+import { balance, balanceZero } from '$lib/derived/balances.derived';
+import { isBusy } from '$lib/derived/busy.derived';
+import { exchangeInitialized, exchanges } from '$lib/derived/exchange.derived';
+import { userHasPouhCredential } from '$lib/derived/has-pouh-credential';
+import { routeNetwork, routeToken } from '$lib/derived/nav.derived';
+import {
+	combinedDerivedSortedNetworkTokens,
+	combinedDerivedSortedNetworkTokensUi
+} from '$lib/derived/network-tokens.derived';
+import {
+	networkAddress,
+	networkBitcoin,
+	networkEthereum,
+	networkICP,
+	networkId,
+	pseudoNetworkChainFusion,
+	selectedNetwork
+} from '$lib/derived/network.derived';
+import { networks, networksMainnets, networksTestnets } from '$lib/derived/networks.derived';
+import { pageToken } from '$lib/derived/page-token.derived';
+import { hideZeroBalances, showZeroBalances, testnetsEnabled } from '$lib/derived/settings.derived';
+import { testnets } from '$lib/derived/testnets.derived';
+import {
+	tokenDecimals,
+	tokenId,
+	tokenStandard,
+	tokenSymbol,
+	tokenToggleable,
+	tokenWithFallback
+} from '$lib/derived/token.derived';
+import {
+	enabledErc20Tokens,
+	enabledIcTokens,
+	enabledMainnetTokensUsdBalancesPerNetwork,
+	enabledTokens,
+	tokens,
+	tokensToPin
+} from '$lib/derived/tokens.derived';
+import { tick } from 'svelte';
+
+const derivedList = {
+	authIdentity,
+	authNotSignedIn,
+	authSignedIn,
+	balance,
+	balanceZero,
+	btcAddressMainnet,
+	btcAddressTestnet,
+	combinedDerivedSortedNetworkTokens,
+	combinedDerivedSortedNetworkTokensUi,
+	enabledErc20Tokens,
+	enabledIcTokens,
+	enabledMainnetTokensUsdBalancesPerNetwork,
+	enabledTokens,
+	exchangeInitialized,
+	exchanges,
+	ethAddress,
+	ethAddressCertified,
+	ethAddressNotCertified,
+	ethAddressNotLoaded,
+	hideZeroBalances,
+	isBusy,
+	networkAddress,
+	networkBitcoin,
+	networkEthereum,
+	networkICP,
+	networkId,
+	networks,
+	networksMainnets,
+	networksTestnets,
+	pageToken,
+	pseudoNetworkChainFusion,
+	routeNetwork,
+	routeToken,
+	selectedNetwork,
+	showZeroBalances,
+	testnets,
+	testnetsEnabled,
+	tokenDecimals,
+	tokenId,
+	tokenStandard,
+	tokenSymbol,
+	tokenToggleable,
+	tokenWithFallback,
+	tokenWithFallbackAsIcToken,
+	tokens,
+	tokensToPin,
+	userHasPouhCredential
+};
+
+export const testDerivedUpdates = async (changeStore: () => void) => {
+	const derivedChangeCounters = new Map<string, number>();
+
+	Object.entries(derivedList).forEach(([name, derivedStore]) => {
```

Comment:
Maybe it's a question of understanding. Will `testDerivedUpdates` be reused or is this function meant for a single test only?

Reply:
ah, it will be reused for all the other stores. Maybe, if you want, I can already put another store in this PR

Reply:
Since this is the PR that will defines the test I think it makes sense to put another store here to get a better picture. 
At least it will help me understood better.

Reply:
Done! Included `balancesStore` test

Reply:
Thanks! Got it.